lib/improver/tests/test_ensemble_copula_coupling_EnsembleCopulaCouplingUtilities.py

```diff
@@ -41,10 +41,161 @@
+            reshape_array_to_have_probabilistic_dimension_at_the_front(
+                cube.data, cube, "percentile", plen))
+        self.assertEqual(reshaped_array.shape[0], plen)
+        self.assertEqual(reshaped_array.shape, (3, 1, 3, 3))
+
+    def test_percentile_is_not_a_dimension_coordinate(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        cube = self.current_temperature_forecast_cube
+        for cube_slice in cube.slices_over("percentile"):
+            break
+        input_array = cube_slice.data
+        plen = len(cube_slice.coord("percentile").points)
+        reshaped_array = (
+            reshape_array_to_have_probabilistic_dimension_at_the_front(
+                cube_slice.data, cube_slice, "percentile", plen))
+
+    def test_missing_coordinate(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        cube = self.current_temperature_forecast_cube
+        input_array = cube.data
+        plen = len(cube.coord("percentile").points)
+        msg = "coordinate is not available"
+        with self.assertRaisesRegexp(CoordinateNotFoundError, msg):
+            reshape_array_to_have_probabilistic_dimension_at_the_front(
+                cube.data, cube, "nonsense", plen)
+
+
+class Test_insert_lower_and_upper_endpoint_to_1d_array(IrisTest):
+
+    """Test the insert_lower_and_upper_endpoint_to_1d_array."""
+
+    def test_basic(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        expected = [0, 0.2, 0.5, 0.8, 1]
+        percentiles = [0.2, 0.5, 0.8]
+        result = insert_lower_and_upper_endpoint_to_1d_array(
+            percentiles, 0, 1)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected)
+
+    def test_another_example(self):
+        """
+        Another basic test that the result is a numpy array with the
+        expected contents.
+        """
+        expected = [-100, -40, 200, 1000, 10000]
+        percentiles = [-40, 200, 1000]
+        result = insert_lower_and_upper_endpoint_to_1d_array(
+            percentiles, -100, 10000)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected)
+
+
+class Test_concatenate_2d_array_with_2d_array_endpoints(IrisTest):
+
+    """Test the concatenate_2d_array_with_2d_array_endpoints."""
+
+    def test_basic(self):
+        """
+        Basic test that the result is a numpy array with the expected contents.
+        """
+        expected = np.array([[0, 0.2, 0.5, 0.8, 1]])
+        percentiles = np.array([[0.2, 0.5, 0.8]])
+        result = concatenate_2d_array_with_2d_array_endpoints(
+            percentiles, 0, 1)
+        self.assertIsInstance(result, np.ndarray)
+        self.assertArrayAlmostEqual(result, expected)
+
+    def test_another_example(self):
```

Comment:
Could this instead contain a 2D cube with more than one percentile in.
 eg
[[0.2, 0.5, 0.8], [0.2, 0.5, 0.8], ]